dao-contracts/src/voting/governance_voting.rs

```diff
@@ -0,0 +1,366 @@
+                };
+                let mut voters = self.voters.get(&voting_id);
+                // Add a voter to the list
+                voters.push(voter);
+                self.voters.set(&voting_id, voters);
+            }
+        }
+
+        // Update the votes list
+        self.votes.set(&(voting_id, voter), vote);
+
+        // update voting
+        voting.stake(stake, choice);
+        self.votings.set(&voting_id, voting);
+
+        emit(VoteCast {
+            voter,
+            voting_id,
+            choice,
+            stake,
+        });
+    }
+
+    pub fn get_dust_amount(&self) -> U256 {
+        self.dust_amount.get()
+    }
+
+    pub fn get_variable_repo_address(&self) -> Address {
+        self.variable_repo.get().unwrap_or_revert()
+    }
+
+    pub fn get_reputation_token_address(&self) -> Address {
+        self.reputation_token.get().unwrap_or_revert()
+    }
+
+    pub fn perform_action(&self, voting: &Voting) {
+        call_contract(
+            voting.contract_to_call().unwrap_or_revert(),
+            voting.entry_point(),
+            voting.runtime_args().clone(),
+        )
+    }
+
+    fn transfer_reputation(&mut self, owner: Address, recipient: Address, amount: U256) {
+        let args: RuntimeArgs = runtime_args! {
+            "owner" => owner,
+            "recipient" => recipient,
+            "amount" => amount,
+        };
+
+        call_contract(self.get_reputation_token_address(), "transfer_from", args)
+    }
+
+    fn burn_reputation(&mut self, owner: Address, amount: U256) {
+        let args: RuntimeArgs = runtime_args! {
+            "owner" => owner,
+            "amount" => amount,
+        };
+
+        call_contract(self.get_reputation_token_address(), "burn", args)
+    }
+
+    fn burn_creators_and_return_others_reputation(&mut self, voting_id: VotingId) {
+        let voters = self.voters.get(&voting_id);
+        for (i, address) in voters.iter().enumerate() {
+            let vote = self.votes.get(&(voting_id, *address));
+            if i == 0 {
+                // the creator
+                self.burn_reputation(self_address(), vote.stake);
+            } else {
+                // the voters - transfer from contract to them
+                self.transfer_reputation(self_address(), *address, vote.stake);
+            }
+        }
+    }
+
+    fn return_reputation(&mut self, voting_id: VotingId) {
+        let voters = self.voters.get(&voting_id);
+        for address in voters.iter() {
+            let vote = self.votes.get(&(voting_id, *address));
+            self.transfer_reputation(self_address(), *address, vote.stake);
+        }
+    }
+
+    fn redistribute_reputation(&mut self, voting: &Voting) {
+        // TODO: remove bytes conversion after support for U256<>U512 conversion will be added to Casper
+        let voters = self.voters.get(&voting.voting_id());
+        let total_stake = U512::convert_from_bytes(voting.total_stake().convert_to_bytes());
```

Comment:
You can't use `BytesConversion` in contracts, as it is implemented using `unwrap`. But it's definitely useful to have it. So change it to:
```
pub trait BytesConversion: Sized {
    fn convert_to_bytes(&self) -> Result<Bytes, Error>;
    fn convert_from_bytes(bytes: Bytes) -> Result<Self, Error>;
}

impl<T: ToBytes + FromBytes> BytesConversion for T {
    fn convert_to_bytes(&self) -> Result<Bytes, Error> {
        Ok(Bytes::from(self.to_bytes()?))
    }

    fn convert_from_bytes(bytes: Bytes) -> Result<Self, Error> {
        let (v, rest) = T::from_bytes(&bytes)?;
        if !rest.is_empty() {
            Err(Error::LeftOverBytes)
        } else {
            Ok(v)
        }
    }
}
``` 
and wrap this behavior `U512::convert_from_bytes(vote.stake.convert_to_bytes())` in a function as it is used a lot.